src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -6126,6 +6126,10 @@ private ExpressionAst GetCommandArgument(CommandArgumentContext context, Token t
                             case CommandArgumentContext.CommandNameAfterInvocationOperator:
                                 token.TokenFlags |= TokenFlags.CommandName;
                                 break;
+                            case CommandArgumentContext.SwitchCondition when token.Kind == TokenKind.Identifier && token.Text.Equals("default", StringComparison.OrdinalIgnoreCase):
+                                // `Default` is a keyword of very limited scope so do color it that way in this case
+                                token.TokenFlags |= TokenFlags.Keyword;
```

Comment:
I believe we need to enhance `enum TokenKind`, `s_staticTokenFlags`, `s_keywordTokenKind`.

Reply:
@iSazonov, are you suggesting I try to implement `default` as a fully specified token?

- [x] Add to `enum.TokenKind`, `s_staticTokenFlags`, `s_tokenText`.
- [x] Add to `s_keywordText`, `s_keywordTokenKind`.
- [x] Correct assertions for table length checks.
- [ ] Use token in `switch` statement condition parsing(`parser`) logic.
- [ ] Implement tests …

It does not appear that I have to add any further logic to tokenizer, correct?

This way has higher difficulty than it might first appear, because now logic that was purposely put to prevent keywords from being detected in this position now has to be bypassed.  Or in other words, the code changes I already committed (seen here) are still needed to some extent, even after adding the `default` keyword to the keyword token list.

Reply:
> to implement default as a fully specified token

I think it was a conclusion in the issue.
@daxian-dbw @rjmholt Could you please share your thoughts before @msftrncs continue?

Reply:
I think I was able to modify the switch condition clause logic to utilize the keyword token, thus preventing the token from being reverted to a command argument, eliminating the need for extra logic in `GetCommandArgument` to prevent that.  However, not sure which way is more efficient.

Reply:
I do have a commit ready to post, implementing `default` as a full token, with revisions to SwitchStatementRule to specifically utilize it instead of a string comparison.  In that case, the change here is no longer needed and such would not be present.